### PR TITLE
MRD-436 Add custodyStatus to recommendation data

### PR DESCRIPTION
**Pull request by jackmaloney**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/jpa/entity/RecommendationEntity.kt [hunk 1]
```diff
@@ -32,17 +33,27 @@ data class RecommendationEntity(
...
+  var custodyStatus: CustodyStatus? = null,
   var status: Status? = null,
   var lastModifiedBy: String? = null,
   var lastModifiedDate: String? = null,
   val createdBy: String? = null,
   val createdDate: String? = null
 ) : Serializable
 
+enum class Recommendation(val text: String) {
+  STANDARD("Standard"), NO_RECALL("No recall"), FIXED_TERM("Fixed term"),
```
Comment (jonnywyatt) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/jpa/entity/RecommendationEntity.kt:45]: I don't know if this impacts you but the labels are going to change slightly in the front end - `Standard recall`, `Fixed term recall`, `No recall`.
Should they be hardcoded here? The source of truth is the UI

Comment (bills7281) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/jpa/entity/RecommendationEntity.kt]: These aren't needed anymore :)
Let me check out your branch Jack to see if it all still works with these removed.

Comment (jackmaloney) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/jpa/entity/RecommendationEntity.kt:45]: Actually, that's a rebase issue - this shouldn't be in there, Bill removed as part of his last PR. I'll fix now...

Comment (bills7281) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/jpa/entity/RecommendationEntity.kt]: ah okay :)

Comment (jonnywyatt) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/jpa/entity/RecommendationEntity.kt]: 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/jpa/entity/RecommendationEntity.kt [hunk 2]
```diff
@@ -43,6 +45,11 @@ enum class Status {
   DRAFT, DOCUMENT_CREATED, DELETED
 }
 
+data class CustodyStatusOption(
```
Comment (jonnywyatt) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/jpa/entity/RecommendationEntity.kt:48]: Minor point, and don't even know if it's the right way to do it in Kotlin, but every recommendation field that choose from options, will use a structure like this. Could use a generic class?

Comment (jackmaloney) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/jpa/entity/RecommendationEntity.kt:48]: Yeh good point, I'll look at putting that in

**Review by bills7281 — APPROVED** [review]

**Review by jonnywyatt — APPROVED** [review]

**Pull request merged**